src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/Unsafe.cs

```diff
@@ -671,7 +671,7 @@ public static void Write<T>(void* destination, T value)
         // Mono:AsRef
         [NonVersionable]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static ref T AsRef<T>(in T source)
+        public static ref T AsRef<T>([LifetimeAnnotation(true, false)] in T source)
```

Comment:
FYI: this is effectively encoding that `AsRef` is the single method for stripping all safety from a `ref` value. It now removes `in` (changes mutability) and `scoped` (changes lifetime). I'm 100% okay with this, it's the design I prefer, however IIRC there was some debate on whether we should do this or have a separate method for each type of safety we strip. Wanted to double check that this debate was had and we're not accidentally just making that decision here.

Reply:
@jkotas Do you have any thoughts on @jaredpar's recollection?

Reply:
The main concern on my end is making unsafe code harder to reason about. There are both pros and cons to having a single method here and I think its overall fine for this scenario.

I think the bigger annoyance on my end is actually the IL bloat and readability issues around having to strip `in` when doing more complex `Unsafe` logic, even where the underlying value isn't mutated, but that's a separate issue and one that is ideally handled by the planned language to allow migration from `ref` to `in` and `ref readonly` without being a binary or source breaking change.